src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -21,25 +32,471 @@ const ActorInit En_Bombf_InitVars = {
+                Audio_PlayActorSound2(&this->actor, NA_SE_PL_PULL_UP_ROCK);
+                this->actor.flags &= ~1;
+            } else {
+                player->actor.attachedB = NULL;
+                player->heldActor = NULL;
+                player->interactRangeActor = NULL;
+                this->actor.attachedA = NULL;
+                player->stateFlags1 &= ~0x800;
+            }
+        } else if ((this->bombCollider.base.acFlags & 2)) {
+            this->bombCollider.base.acFlags &= ~2;
+
+            if (this->bombCollider.base.ac->type != ACTORTYPE_BOSS) {
+                bomb = (EnBombf*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOMBF, this->actor.posRot.pos.x,
+                                             this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0);
+                if (bomb != NULL) {
+                    bomb->unk_200 = 1;
+                    bomb->timer = 0;
+                    this->timer = 180;
+                    this->actor.flags &= ~1;
+                    this->flowerBombScale = 0.0f;
+                }
+            }
+        } else {
+            if (func_8008EF5C(globalCtx, &this->actor.posRot.pos, 30.0f, 50.0f)) {
+                bomb = (EnBombf*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOMBF, this->actor.posRot.pos.x,
+                                             this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0);
+                if (bomb != NULL) {
+                    bomb->timer = 100;
+                    this->timer = 180;
+                    this->actor.flags &= ~1;
+                    this->flowerBombScale = 0.0f;
+                }
+            } else {
+                if (!func_8002F410(&this->actor, globalCtx)) {
+                    func_8002F580(&this->actor, globalCtx);
+                } else {
+                    player->actor.attachedB = NULL;
+                    player->heldActor = NULL;
+                    player->interactRangeActor = NULL;
+                    this->actor.attachedA = NULL;
+                    player->stateFlags1 &= ~0x800;
+                    this->actor.posRot.pos = this->actor.initPosRot.pos;
+                }
+            }
+        }
+    } else {
+        if (this->timer == 0) {
+            this->flowerBombScale += 0.05f;
+            if (this->flowerBombScale >= 1.0f) {
+                this->actor.flags |= 1;
+            }
+        }
+
+        if (func_8002F410(&this->actor, globalCtx)) {
+            player->actor.attachedB = NULL;
+            player->heldActor = NULL;
+            player->interactRangeActor = NULL;
+            this->actor.attachedA = NULL;
+            player->stateFlags1 &= ~0x800;
+            this->actor.posRot.pos = this->actor.initPosRot.pos;
+        }
+    }
+}
+
+void EnBombf_Move(EnBombf* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx)) {
+        this->flowerBombScale = 0.0f; // this doesnt do anything 
```

Comment:
What do you mean by these comments that say "this doesnt do anything"? I see the variable is used so surely it at least has a purpose, even if it's not "relevant" under normal conditions or something like that.

Reply:
the bomb flower scale is only relevant for the actual bomb flower that the bomb grows in. This function only runs for the bomb that actually explodes, so this just seems to be a left over. Modifying bomb flower scale has 0 effect on the bomb that actually ends up exploding

Reply:
if you want me to be more clear about hat in the comments i can do that

Reply:
I see, then yeah I think the comments could be a bit clearer on this matter. Specifically on the fact that it has no effect in the context of exploding bombs.